3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/CurrencyConversionEnrichmentSpec.scala

```diff
@@ -0,0 +1,63 @@
+  val currencyInvalidHul = """Open Exchange Rates error, message: Currency [HUL] is not supported by Joda money Currency not found in the API, invalid currency , Open Exchange Rates error, message: Currency [HUL] is not supported by Joda money Currency not found in the API, invalid currency , Open Exchange Rates error, message: Currency [HUL] is not supported by Joda money Currency not found in the API, invalid currency , Open Exchange Rates error, message: Currency [HUL] is not supported by Joda money Currency not found in the API, invalid currency"""
+  def e1 =
+    "SPEC NAME"                    || "TRANSACTION CURRENCY"   | "API Key"                                  | "TRANSACTION AMOUNT"                       | "TRANSACTION ITEM CURRENCY"     | "TRANSACTION ITEM PRICE"                |"CONVERTED TUPLE"                                                          |
+    "First Success Test"           !! "GBP"                    ! appId                                      ! TransactionAmounts("12.99", "2.67", "0")   ! "GBP"                           ! "12.99"                                 !(TransactionAmounts("8.65","1.78","0.0"),"8.65").success                   |
+    "First Failure Test"           !! "RUP"                    ! appId                                      ! TransactionAmounts("14.99", "1.17", "0")   ! "GBP"                           ! "17.99"                                 ! currencyInvalidRup.failureNel                                             |
+    "Second Failure Test"          !! "HUL"                    ! appId                                      ! TransactionAmounts("10.99", "0.7",  "0")   ! "GBP"                           ! "1.99"                                  ! currencyInvalidHul.failureNel                                             |
+    "Third Failure Test"           !! "GBP"                    ! "8A8A8A8A8A8A8A8A8A8A8A8AA8A8A8A8"         ! TransactionAmounts("2.99",  "3.67", "0")   ! "GBP"                           ! "2.99"                                  ! appIdInvalid.failureNel                                                   |
+    "Second Success Test"          !! "USD"                    ! appId                                      ! TransactionAmounts("11.09", "4.67", "2")   ! "GBP"                           ! "10.99"                                 !(TransactionAmounts("11.09","4.67","2.0"),"7.32").success                  |
+    "Third Success Test"           !! "GBP"                    ! appId                                      ! TransactionAmounts("10.00", "2.67", "0")   ! "GBP"                           ! "10.00"                                 !(TransactionAmounts("6.66","1.78","0.0"),"6.66").success                   |> {
+      (_, trCurrency, apiKey, trAmount, tiCurrency, tiPrice, expected) =>
+        CurrencyConversionEnrichment(apiKey, "EUR", "EOD_PRIOR").convertCurrencies(trCurrency, trAmount, tiCurrency, tiPrice) must_== expected
+    }
+
```

Comment:
Move your failure tests into a def e2. No need to have successes and failures in the same test.